### DISCUSSION TITLE
Pressure fluctuation - HVAC

### DISCUSSION THREAD

**Discussion by Er9y714** [firemodels/fds#13629] [General]
Hi,

I am posting this to discussions instead of issues because a similar issue was discussed before (#11951) and the behaviour was deemed normal. So this time I would like to share an another setup (DUCT with HVAC) and ask your opinion. Is this normal, if yes, how can I prevent it? Thank you.

I have observed a pressure issue in a complex simulation I am working on. In short, a room is connected to outside via a HVAC duct. When all doors are closed, the pressure in the room with duct does not drop to 0 Pa as I would expect. 

I have tried to simplify the case as much as possible. I have also tightened the tolerances to the level I cannot apply to the complete project file. Yet, the issue is still present. 
Although it is not an option for large project file, I have tried running this small case with ULMAT as well. It was significantly slower and fluctuation was still there. 

(Similar issue #11951) Based on the previous posts/issues, I am aware that closing a door suddenly is expected to create such fluctuations but I have not yet found an alternative door opening/closing method. Furthermore, the average of the pressure in the room when all doors are closed is positive and not reducing. With a duct of 1.5 m2 area, I would expect there is not sufficient air supply to balance this pressure fluctuation but it appears not the case. 

I have tried running this small case with ULMAT as well. It was significantly slower and fluctuation was still there. (This solver is not an option to use in complex project file)

Similar issue #11951

![image](https://github.com/user-attachments/assets/6cfe1018-6113-438a-aee0-5bbabcafecb3)
![image](https://github.com/user-attachments/assets/92c42a00-085f-42dc-83a3-0bfd90711e21)

```

&HEAD CHID='test_3'/
&TIME T_END=9.0/
&DUMP DT_DEVC=0.01, DT_SLCF=0.1/
&MISC MINIMUM_ZONE_VOLUME=0.5/
&PRES MAX_PRESSURE_ITERATIONS =100, PRESSURE_TOLERANCE=10, VELOCITY_TOLERANCE=0.01/

&MESH ID='MESH_006-merged-merged-02', IJK=28,16,42, XB=28.4,31.2,42.6,44.2,0.0,4.2/
&MESH ID='MESH_006-merged-merged-03', IJK=28,16,42, XB=28.4,31.2,44.2,45.8,0.0,4.2/
&MESH ID='MESH_006-merged-merged-04', IJK=28,16,42, XB=28.4,31.2,45.8,47.4,0.0,4.2/
&MESH ID='MESH_006-merged-merged-05', IJK=28,16,42, XB=28.4,31.2,47.4,49.0,0.0,4.2/
&MESH ID='MESH_008', IJK=30,22,20, XB=25.4,28.4,44.2,46.4,2.2,4.2/
&MESH ID='MESH_011', IJK=26,33,22, XB=25.4,28.0,43.8,47.1,4.2,6.4/
&MESH ID='MESH_013', IJK=30,32,20, XB=25.4,28.4,41.0,44.2,2.2,4.2/
&MESH ID='MESH_014', IJK=30,22,22, XB=25.4,28.4,44.2,46.4,0.0,2.2/

&RAMP ID='CTRL lobby door 1_RAMP', T=-5.25, F=-1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=-4.75, F=1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=2.75, F=1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=3.25, F=-1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=5.75, F=-1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=6.25, F=1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=8.75, F=1.0/
&RAMP ID='CTRL lobby door 1_RAMP', T=9.25, F=-1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=-5.25, F=1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=-4.75, F=-1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=3.75, F=-1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=4.25, F=1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=6.75, F=1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=7.25, F=-1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=9.75, F=-1.0/
&RAMP ID='CTRL lobby door 2_RAMP', T=10.25, F=1.0/
&CTRL ID='CTRL lobby door 1', FUNCTION_TYPE='CUSTOM', RAMP_ID='CTRL lobby door 1_RAMP', LATCH=.FALSE., INPUT_ID='TIME'/
&CTRL ID='CTRL lobby door 2', FUNCTION_TYPE='CUSTOM', RAMP_ID='CTRL lobby door 2_RAMP', LATCH=.FALSE., INPUT_ID='TIME'/
&DEVC ID='Core2_Door01_VolFlow', QUANTITY='V-VELOCITY', SPATIAL_STATISTIC='AREA INTEGRAL', XB=29.0,29.9,46.45,46.45,-5.551115E-17,2.1/
&DEVC ID='Core2_DoorStairs02_VolFlow', QUANTITY='U-VELOCITY', SPATIAL_STATISTIC='AREA INTEGRAL', XB=28.455874,28.455874,45.2,46.1,0.0,2.1/
&DEVC ID='Louvre_VolFlow', QUANTITY='NORMAL VELOCITY', SPATIAL_STATISTIC='SURFACE INTEGRAL', XB=25.7,27.0,46.4,46.4,4.4,5.6, IOR=-2/
&DEVC ID='Louvre_VolFlow01', QUANTITY='NORMAL VELOCITY', SPATIAL_STATISTIC='SURFACE INTEGRAL', XB=25.7,27.0,46.8,46.8,4.4,5.6/
&DEVC ID='Core2_Door01_MassFlow', QUANTITY='MASS FLUX Y', SPATIAL_STATISTIC='AREA INTEGRAL', XB=29.0,29.9,46.45,46.45,-5.551115E-17,2.1/
&DEVC ID='Core2_DoorStairs02_MassFlow', QUANTITY='MASS FLUX X', SPATIAL_STATISTIC='AREA INTEGRAL', XB=28.455874,28.455874,45.2,46.1,0.0,2.1/
&DEVC ID='Louvre_MassFlow', QUANTITY='MASS FLUX WALL', SPATIAL_STATISTIC='SURFACE INTEGRAL', XB=25.7,27.0,46.4,46.4,4.4,5.6/
&DEVC ID='Louvre_MassFlow01', QUANTITY='MASS FLUX', SPATIAL_STATISTIC='SURFACE INTEGRAL', XB=25.7,27.0,46.8,46.8,4.4,5.6/
&DEVC ID='P_10', QUANTITY='PRESSURE', XYZ=29.95,43.991975,2.0/
&DEVC ID='P_11', QUANTITY='PRESSURE', XYZ=26.959199,44.968723,2.0/
&DEVC ID='P_12', QUANTITY='PRESSURE', XYZ=27.648236,45.851082,5.2/
&DEVC ID='TIME', QUANTITY='TIME', XYZ=28.4,42.6,0.0/

&SURF ID='ADIABATIC',
      COLOR='GRAY 80',
      DEFAULT=.TRUE.,
      ADIABATIC=.TRUE./

&OBST ID='wall', XB=28.4,28.6,42.6,44.2,0.0,4.2/ 
&OBST ID='wall', XB=28.6,30.2,42.6,44.2,4.1,4.2/ 
&OBST ID='wall', XB=30.2,31.2,42.6,44.2,0.0,4.2/ 
&OBST ID='wall', XB=28.4,28.6,44.2,45.2,0.0,4.2/ 
&OBST ID='wall', XB=28.4,28.6,45.2,45.8,2.1,4.2/ 
&OBST ID='wall', XB=28.6,30.2,44.2,45.8,4.1,4.2/ 
&OBST ID='wall', XB=30.2,31.2,44.2,45.8,0.0,4.2/ 
&OBST ID='wall', XB=28.4,28.6,46.1,46.4,0.0,4.2/ 
&OBST ID='wall', XB=28.4,28.6,45.8,46.1,2.1,4.2/ 
&OBST ID='wall', XB=28.4,29.0,46.4,46.8,0.0,4.2/ 
&OBST ID='wall', XB=28.6,30.2,45.8,46.4,4.1,4.2/ 
&OBST ID='wall', XB=29.0,29.9,46.4,46.6,2.1,4.2/ 
&OBST ID='wall', XB=29.9,31.2,46.4,46.8,0.0,4.2/ 
&OBST ID='wall', XB=30.2,31.2,45.8,46.4,0.0,4.2/ 
&OBST ID='wall', XB=28.2,28.4,46.1,46.4,2.2,4.2/ 
&OBST ID='wall', XB=28.2,28.4,46.1,46.4,0.0,2.2/ 
&OBST ID='wall', XB=25.4,28.0,46.4,46.8,4.2,6.4/ 
&OBST ID='door1', XB=29.0,29.9,46.4,46.5,0.0,2.1, SURF_ID='ADIABATIC', CTRL_ID='CTRL lobby door 1'/ 
&OBST ID='door2', XB=28.4,28.5,45.2,45.8,0.0,2.1, SURF_ID='ADIABATIC', CTRL_ID='CTRL lobby door 2'/ 
&OBST ID='door2', XB=28.4,28.5,45.8,46.1,0.0,2.1, SURF_ID='ADIABATIC', CTRL_ID='CTRL lobby door 2'/ 

&VENT ID='Vent04', SURF_ID='OPEN', XB=25.4,28.0,47.1,47.1,4.2,6.4/ 
&VENT ID='Vent05', SURF_ID='OPEN', XB=28.4,31.2,49.0,49.0,0.0,4.2, DYNAMIC_PRESSURE=-100.0/ 
&VENT ID='Louvre_node01', SURF_ID='HVAC', XB=25.7,27.0,46.39,46.39,4.4,5.6, COLOR='GREEN YELLOW'/ 
&VENT ID='Louvre_node02', SURF_ID='HVAC', XB=25.7,27.0,46.81,46.81,4.4,5.6, COLOR='GREEN YELLOW'/ 

&HVAC ID='Duct01', TYPE_ID='DUCT', AREA=1.49, PERIMETER=5.0, LOSS=0.25,0.25, NODE_ID='Node01','Node02', ROUGHNESS=1.0E-3, LENGTH=0.42/
&HVAC ID='Node02', TYPE_ID='NODE', DUCT_ID='Duct01', VENT_ID='Louvre_node02', XYZ=26.3,46.4,5/
&HVAC ID='Node01', TYPE_ID='NODE', DUCT_ID='Duct01', VENT_ID='Louvre_node01', XYZ=26.3,46.4,5/

&SLCF QUANTITY='PRESSURE', ID='Slice', PBZ=2.865675/


&TAIL /

```

**Comment by mcgratta** [comment]
I'll take a look.

**Comment by drjfloyd** [comment]
If I delete the HVAC lines the odd pressure behavior still exists. I think this is a result of having a large negative dynamic pressure at one of the open boundaries that draws a vacuum in its mesh until doors open to the structure.

**Comment by mcgratta** [comment]
I replaced the green vent with a `HOLE` and the pressure looks like this:

![image](https://github.com/user-attachments/assets/c4228a03-fdca-47da-bcdc-7849610810a1)

**Comment by mcgratta** [comment]
Could it be that a large cross section, short length, low resistance `HVAC` duct might oscillate like this?

**Comment by drjfloyd** [comment]
The LOSS is low for a louver. A louvered opening in a wall is going act like an orifice in a wall with a discharge coefficient (C_d) of ~0.6 to ~0.7.  Loss (K) = 1/C_d^2 or K ~2.04 to ~2.78. 



<img width="449" alt="image" src="https://github.com/user-attachments/assets/afdb1e94-f9f1-45fa-9e95-e226685f2e88">

**Comment by Er9y714** [comment]
The loss on the louvre belongs to only the louvre element (excluding inlet and outlet losses, since they are calculated by CFD)(~90% opening rate). 


The reason of using DUCT instead of a HOLE is DUCT gives me the opportunity to adjust the LOSS exactly the value I enter. On the other hand, this project started with 50% opening rate with higher loss. Later it changed to 90% opening rate with low loss value. That is why I encountered this setup. 

All in all, is this issue linked to low loss value? Shall I check if it changes with higher loss value?

**Reply by drjfloyd** [reply]
I suggest you try what the User's Guide suggests for HVAC stabilty issues. First try setting HVAC_PRES_RELAX to something less than one.

**Reply by Er9y714** [reply]
Thank you. 

With added line:  HVAC_LOCAL_PRESSURE = .FALSE.
![image](https://github.com/user-attachments/assets/e68bf647-83b0-4a45-a794-16571dde49fe)


With added line:  HVAC_PRESS_RELAX=0.01:
![image](https://github.com/user-attachments/assets/4179a9c5-62b4-4fbd-9321-1369500d2096)

With added line:  HVAC_PRESS_RELAX=0.5:
![image](https://github.com/user-attachments/assets/b3fd91c6-821c-4dc0-ad89-dc8eee820fb3)




With added line:  HVAC_PRESS_RELAX=0.9:
![image](https://github.com/user-attachments/assets/cc1df290-9d81-4230-9286-f84fa9bec400)



With added line:  HVAC_PRESS_RELAX=0.99:
![image](https://github.com/user-attachments/assets/1c82f0b2-6228-4d6f-b0b6-b87159f6a9dd)



Increased the DUCT length to 3 m and loss to 10:
 &HVAC ID='Duct01', TYPE_ID='DUCT', AREA=1.49, PERIMETER=5.0, LOSS=10,10, NODE_ID='Node01','Node02', ROUGHNESS=1.0E-3, LENGTH=3/
![image](https://github.com/user-attachments/assets/29f7b21f-d254-48db-98cb-c59fbf58789b)
 


Furthermore, may be not relevant, I wanted tested the mass transport option. But I encountered a crash when setting N_CELLS to 200. 

With added line: N_CELLS=200 (at the duct) the simulation **crashes at 3.24 s** (run with parameters -o 1 -p 8). It gives **severe(157)** error. 

With added line: N_CELLS=10 (at the duct) the simulation does not crash. But the fluctuation is still there. 
![image](https://github.com/user-attachments/assets/81a56dc4-1070-4dc3-b7a2-9fd4b968e637)

**Comment by drjfloyd** [comment]
HVAC_MASS_TRANSPORT does not change how the duct pressure solution is done. It just adds transport delays.  Since your case is all ambient temperature, I would not expect to see significant differences with HVAC_MASS_TRANSPORT.  What duct length crashed with 200 nodes? 

You are imposing a 100 Pa pressure drop on the interior side of the louver. With a louver loss of 0.25, the steady velocity is ~26 m/s. That will wind up being ~600 air changes per hour inside the rooms. This may be too challenging of a condition for this geometry and HVAC. What is that 100 Pa pressure meant to represent? Is it realistic that such a pressure drop would be maintained once a path is opened up to the louvers?

**Reply by Er9y714** [reply]
The pressure boundary was set for demonstration of the case. 100 Pa was not a representation of another system. I updated it to -1 Pa. The fluctuation persists. 
![image](https://github.com/user-attachments/assets/d98003a6-f0b4-4cb2-b324-24888d6cf294)



The crashing case had the duct length of 0.42 m as the original case (LENGTH=0.42).

**Reply by drjfloyd** [reply]
I commented out the Lobby Door 2 OBST so it is always open.  Then with HVAC_PRES_RELAX=0.8 and no &PRES line I get the plot below for the pressure which is much more reasonable. 

![image](https://github.com/user-attachments/assets/ef01f950-e5d9-4664-ab28-22c5a2a84898)

If instead I comment out Lobby Door 1 OBST:

![image](https://github.com/user-attachments/assets/5d79d223-7a49-4758-bd27-d459a8ee1d94)

It is with both doors that things go off the rails. It even happens if there is no DYNAMIC_PRESSURE. Testing at our large scale facility this week and next so will have limited time to delve more into this right now.

**Comment by Er9y714** [comment]
Thank you very much for the update. We have continued on the project with a workaround. Meanwhile, I will continue to follow up this thread. Please let me know if you would like me to try any setup.

**Comment by mcgratta** [comment]
I'll work the issue too. One thing to try is to apply leakage to the walls, and I would avoid the use of `ADIABATIC`. Not that it matters in this case, but in general, it's not a good boundary condition when instabilities are occurring.

**Reply by Er9y714** [reply]
Noted. When I was simplifying the case, I created it. The original case in the project is not Adiabatic.

**Comment by mcgratta** [comment]
If I set the `DYNAMIC_PRESSURE` back to 0 and save the output every time step, I get the following "ringing" of the pressure and volume flux at the louvered vent. The slightest pressure change induces a volume flow, that eventually reverses the pressure, and eventually reverses the volume flow. And so on. This is caused by the fact that the pressure solver is not coupled with the HVAC solver. 

![image](https://github.com/user-attachments/assets/90f5143d-d7c3-477d-aa0d-cf79e5cc5339)

**Reply by drjfloyd** [reply]
The two plots I gave in my prior post were only having one each of the two doors. For those things behave OK.  The riniging dampens out fairly quickly. Something with the second pressure zone being broken is getting things to go unstable. There might be some underlying issue as with no dynamic pressure and everything set to the same initial pressure and stratification there shouldn't be anything to cause a change in pressure with two zones that isn't happening with just one zone.

**Reply by mcgratta** [reply]
I suspect that the initial quiet period is an unstable equilibrium. A small pressure perturbation due to opening/closing doors might be just what is needed to start the oscillations. Notice in my plot they start small and keep growing.

**Comment by mcgratta** [comment]
You may be doing this already, but if not, try adding lines like this to your input file
```
&ZONE XYZ=26,45,5, LEAK_AREA=0.1 /
&VENT XB=25.4,25.4,44.0,45.0,5.0,6.0, SURF_ID='leaky wall' /
&SURF ID='leaky wall', LEAK_PATH=1,0, COLOR='GRAY 50' /
```
This creates a leak in the wall near the louvered vent. This one is quite large, like 0.3 m by 0.3 m. You can experiment with different sizes. 

A longer term fix to this problem would be to couple the louvered vent with the overall pressure solver for the entire domain.

**Reply by Er9y714** [reply]
When I use this type of leak and the following parameters, the result is more reasonable (not diverging as before, despite the fluctuation between 4-7 s where the doors is closed).



MISC	HVAC_PRES_RELAX	0.8	
PRES	MAX_PRESSURE_ITERATIONS	100	
PRES	PRESSURE_TOLERANCE	10	
PRES	VELOCITY_TOLERANCE	0.01	

![image](https://github.com/user-attachments/assets/3ae0e10a-c7b7-41bc-bf79-d9ce13e9991e)

**Comment by mcgratta** [comment]
In your larger model, does it make sense to include leakage? Unless the facility is specially designed, it is going to leak. These leaks can help to dampen pressure oscillations, both real and numerical.

**Reply by Er9y714** [reply]
In the larger model, every door has leakages. Therefore, there is not any sealed zone without leakages in the model. Despite them, the model had this abovementioned issue. Then I removed those door leakages when I was simplifying and investigating the issue.

**Comment by drjfloyd** [comment]
I think I found a possible issue with how I deal with connected zones in HVAC; however, for this case that issue would not have any impact. I did; however, discover another possible issue.  If I run this case with one MPI process things look OK. I took your case, removed the DUMP line to just get all the output and added a DEVC for the inside node pressure.

&DEVC QUANTITY='NODE PRESSURE',NODE_ID='Node01',ID='N1'/

With one process (n=1, blue solid line) the oscillations dampen out quickly.  With 8 (n=8 orange dotted line) the oscillations stay.

![image](https://github.com/user-attachments/assets/e0fd95f2-b5fb-42fb-b8a6-0e1bc3631283)

**Reply by Er9y714** [reply]
I am happy to hear you have identified some issues that might help this issue. With the hope of helping in your investigation, an MPI issue was mentioned in my [previous post](https://github.com/firemodels/fds/issues/11951#issuecomment-1624273111) but it had not lead to a fruitful outcome back then.

**Reply by mcgratta** [reply]
Yes, this is a very difficult bug to fix. In theory, there should be no difference in the results if you use one vs several MPI processes, assuming the meshes and other inputs are the same. But when you use one process, the meshes are computed serially, and there might be some subtle difference in outcome when updated  variables in mesh 1 are "seen" when updating mesh 2.

**Reply by Er9y714** [reply]
I appreciate your time. Nowadays, due to the large domains and details we focus on, we never have any project where MPI process is not used. Furthermore, in a simple corridor ventilation study, often 10+ doors are involved. Considering each door having 5+ leak paths, we are always subject to the risk of having this pressure fluctuation issue. That is why this issue is not trivial, at least for me.

**Reply by mcgratta** [reply]
These issues are very common due to the low Mach assumption and the pressure solver. In many instances, the user does not notice these effects, but in cases like yours, they are hard to ignore.

**Comment by mcgratta** [comment]
Did you use the same input file for both cases, just changing the number of MPI processes?

**Reply by drjfloyd** [reply]
It is basically the file in the first post. Changed outputs sllightly. Then ran with 1 or 8 processes.

[test_3.txt](https://github.com/user-attachments/files/17592767/test_3.txt)

**Comment by mcgratta** [comment]
If so, can you post the exact input file you used.

**Comment by mcgratta** [comment]
I am creating an [issue](https://github.com/firemodels/fds/issues/13673) for the case where the results are different depending on the number of MPI processes. I'll keep the discussion open in case there are other aspects of this problem that fall outside that narrow issue.
